File: frontend/src/objects/usergroup/request/AddUserGroupMemberRequest.ts
import type { Username } from '@/objects/user/Username'
import type { NewUserGroupMemberRole } from '@/objects/usergroup/request/NewUserGroupMemberRole'

/** 添加用户组成员请求体；目标用户组由 API path 指定。 */
export type AddUserGroupMemberRequest = {
  username: Username
  role: NewUserGroupMemberRole
}

File: backend/src/main/scala/domains/usergroup/objects/request/AddUserGroupMemberRequest.scala
package domains.usergroup.objects.request

import domains.user.objects.Username
import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

/** 添加用户组成员请求，包含目标用户名和初始成员角色。 */
final case class AddUserGroupMemberRequest(
  username: Username,
  role: NewUserGroupMemberRole
)

/** 提供添加成员请求 JSON 编解码。 */
object AddUserGroupMemberRequest:
  given Encoder[AddUserGroupMemberRequest] = deriveEncoder[AddUserGroupMemberRequest]
  given Decoder[AddUserGroupMemberRequest] = deriveDecoder[AddUserGroupMemberRequest]
